yarn-project/noir-protocol-circuits/src/crates/private-kernel-lib/src/private_kernel_ordering.nr

```diff
@@ -532,4 +536,63 @@ mod tests {
         builder.previous_kernel.end.new_nullifiers = BoundedVec::new(SideEffectLinkedToNoteHash::empty());
         builder.failed();
     }
+
+    #[test]
+    unconstrained fn partitions_data() {
+        let mut builder = PrivateKernelOrderingInputsBuilder::new();
+
+        let mut sorted_new_commitments = [SideEffect::empty(); 10];
+        let mut sorted_new_nullifiers = [SideEffectLinkedToNoteHash::empty(); 10];
+
+        let mut counter = 1;
+        for i in 0..10 {
+            sorted_new_commitments[i] = SideEffect { value: (i + 1) as Field, counter: counter };
+            counter += 1;
+            sorted_new_nullifiers[i] = SideEffectLinkedToNoteHash { value: (i + 11) as Field, counter: counter, note_hash: 0 };
+            counter += 1;
+        }
+
+        for i in 0..10 {
+            builder.previous_kernel.end.new_commitments.push(sorted_new_commitments[9 - i]);
+            builder.previous_kernel.end.new_nullifiers.push(sorted_new_nullifiers[9 - i]);
+        }
+
+        builder.meta_hwm = 10;
+
+        let public_inputs = builder.execute();
+
+        let sorted_unique_commitments = compute_unique_siloed_commitments(
+            // NOTE(fees) changed the tx nullifier here for this test
+            // TODO(fees) review this after #4334
+            public_inputs.end_meta.new_nullifiers[0].value,
+            sorted_new_commitments
+        );
+
+        for i in 0..5 {
+            let meta_commitment = public_inputs.end_meta.new_commitments[i];
+            let commitment = sorted_unique_commitments[i];
+
+            assert(meta_commitment.eq(commitment));
+        }
+
+        // TODO(fees) do something about the magic indices in this test 🙈
+        // start offset by 1 because 0th nullifier is the tx hash
+        for i in 1..5 {
+            let meta_nullifier = public_inputs.end_meta.new_nullifiers[i];
+            let nullifier = sorted_new_nullifiers[i - 1];
+            assert(meta_nullifier.eq(nullifier));
```

Comment:
I'm unclear on why `sorted_new_nullifiers[0] == public_inputs.end_meta.new_nullifiers[1]` because I don't see where the `public_inputs.end_meta.new_nullifiers` are shifted by 1 to make room for the tx hash.

Reply:
@just-mitch and I looked it up and discovered it's injected by the init kernel and for testing purposes, there's this block of code in the builder
https://github.com/AztecProtocol/aztec-packages/blob/bc13c0220823504ecc3162e67e3cf3b41d60b018/yarn-project/noir-protocol-circuits/src/crates/types/src/tests/previous_kernel_data_builder.nr#L47-L52